### PR TITLE
[components] Generate logic for DefinitionsComponent

### PR DESCRIPTION
**Pull request by schrockn**
## Summary & Motivation

Adding basic scaffolding logic for this one. Want to discuss what the right testing harness is for stuff because it is potentially a huge pain.

## How I Tested These Changes

Manual

### CODE REVIEW

**Comment by schrockn** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26657?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#26657** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26657?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26657?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#26656** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26656?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: python_modules/libraries/dagster-components/dagster_components/lib/definitions_component.py [hunk 1]
```diff
@@ -43,7 +44,21 @@ def build_defs(self, context: ComponentLoadContext) -> Definitions:
...
+        generate_params = (
+            params if isinstance(params, DefinitionsGenerateParams) else DefinitionsGenerateParams()
+        )
+
+        with pushd(str(request.component_instance_root_path)):
+            Path(
+                generate_params.definitions_path
+                if generate_params.definitions_path
+                else "definitions.py"
+            ).touch(exist_ok=True)
```
Comment (smackesey) [inline python_modules/libraries/dagster-components/dagster_components/lib/definitions_component.py:57]: Why `exist_ok=True`? This command is designed to run once against an empty directory right?

**Review by smackesey — APPROVED** [review]

File: python_modules/libraries/dagster-components/dagster_components/lib/definitions_component.py (hunk 1 above)
Comment (schrockn) [reply python_modules/libraries/dagster-components/dagster_components/lib/definitions_component.py:57]: Allowing for the use case of it loading from definitions in a file that already exists.

**Comment by schrockn** [comment]
### Merge activity

* **Dec 21, 4:33 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/26657).

**Pull request merged**